### PR TITLE
DEV-1361: User typeahead

### PR DESCRIPTION
**Pull request by alburde1**
**High level description:**
Creating a user typeahead to use for the submission dashboard filter.

**Technical details:**
Endpoint to gather all users that have a submission that the requesting user has at least read access to, split by either FABS or DABS submissions.

**Link to JIRA Ticket:**
[DEV-1361](https://federal-spending-transparency.atlassian.net/browse/DEV-1361)

The following are ALL required for the PR to be merged:
- [x] Backend review completed
- [x] Unit & integration tests updated with relevant test cases
- N/A Frontend impact assessment completed

### CODE REVIEW

File: dataactbroker/handlers/account_handler.py [hunk 1]
```diff
@@ -470,3 +470,57 @@ def list_user_emails():
...
+    """ List user IDs and names that have submissions that the requesting user can see.
+
+        Arguments:
+            d2_submission: boolean indicating whether it is a DABS or FABS submission (True if FABS)
+
+        Returns:
+            A JsonResponse containing a list of users that have submissions that the requesting user can see
+    """
+
+    # print(sess.query(Submission).filter(Submission.user_id == 2).exists())
```
Comment (dpb-bah) [inline dataactbroker/handlers/account_handler.py]: Should this be logged? Or just simply removed?

File: dataactbroker/handlers/account_handler.py [hunk 2]
```diff
@@ -470,3 +470,57 @@ def list_user_emails():
...
+    sess = GlobalDB.db().session
+    # subquery to create the EXISTS portion of the query
+    exists_query = sess.query(Submission).filter(Submission.user_id == User.user_id,
+                                                 Submission.d2_submission.is_(d2_submission))
+
+    # if user is not an admin, we have to adjust the exists query to limit submissions
+    if not g.user.website_admin:
+        # split affiliations into frec and cgac
+        cgac_affiliations = [aff for aff in g.user.affiliations if aff.cgac]
```
Comment (dpb-bah) [inline dataactbroker/handlers/account_handler.py]: Stupid question: if `d2_submission` is true and we're requesting FABS submissions, do we want to filter these with the following:

`cgac_affiliations = [aff for aff in g.user.affiliations if FABS_PERMISSION_ID_LIST]`

I noticed that the FABS_PERMISSION_ID_LIST only works with `submitter`, `editfabs`, and `fabs` whereas `DABS_PERMISSION_ID_LIST` only works with `reader` and `writer`.

Comment (alburde1) [reply dataactbroker/handlers/account_handler.py]: DABS users can view FABS submissions for their agencies but not the other way around (FABS users can't view DABS submissions without DABS permissions)

File: dataactbroker/handlers/account_handler.py (hunk 1 above)
Comment (alburde1) [reply dataactbroker/handlers/account_handler.py]: Oops, fixing

**Pull request merged**